src/nsrt_learning/strips_learning/base_strips_learner.py

```diff
@@ -211,20 +211,20 @@ def _find_best_matching_pnad_and_sub(
         segment: Segment,
         objects: Set[Object],
         pnads: List[PartialNSRTAndDatastore],
-        ground_eff_subset_necessary_eff: bool = False
+        check_only_add_effects: bool = False
     ) -> Tuple[Optional[PartialNSRTAndDatastore], Optional[Dict[Variable,
                                                                 Object]]]:
         """Find the best matching PNAD (if any) given our rationality-based
         score function, and return the PNAD and substitution necessary to
         ground it. If no PNAD from the input list matches the segment, then
         return Nones.
 
-        If ground_eff_subset_necessary_eff is True, then we are calling
-        this function from backchaining during either specialization or
-        spawning a new PNAD and trying to find a substitution that matches
-        a subset of the necessary add effects. If not, we are trying to find
-        a substitution that matches all the necessary add effects (if they
-        have been specified in the segment).
+        If check_only_add_effects is True, we want to find a grounding that
+        is some superset of the necessary_add_effects and also such that
+        the ground operator's add effects are always true in the
+        segment's final atoms. Otherwise, we want to disregard any
+        checks on add effects entirely (we will only make this call when
```

Comment:
when you say "make this call" can you be explicit about whether that's check_only_add_effects = True or check_only_add_effects = False?

Reply:
(i think it's check_only_add_effects = True)